src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -209,7 +209,7 @@ MultiTaskTrackerExecute(Job *job)
 	 * assigning and checking the status of tasks. The second (temporary) hash
 	 * helps us in fetching results data from worker nodes to the master node.
 	 */
-	List *workerNodeList = ActivePrimaryWorkerNodeList(NoLock);
+	List *workerNodeList = ActivePrimaryNonCoordNodeList(NoLock);
```

Comment:
For example, now this line shows how we will fail when coordinator is added as a worker but we won't create task tracker for it.

Reply:
I didn't check but if `ActivePrimaryWorkerNodeList` returns coordinator when `shouldHaveShards=true` for coordinator, it wouldn't make much sense to rename it to `ActivePrimaryNonCoordNodeList`, could you check it ?

Reply:
```c
List *
ActivePrimaryWorkerNodeList(LOCKMODE lockMode)
{
	EnsureModificationsCanRun();
	return FilterActiveNodeListFunc(lockMode, NodeIsPrimaryWorker);
}

static bool
NodeIsPrimaryWorker(WorkerNode *node)
{
	return !NodeIsCoordinator(node) && NodeIsPrimary(node);
}
```